sky/data/data_utils.py

```diff
@@ -126,6 +146,175 @@ def verify_gcs_bucket(name: str) -> bool:
+        if account.name == storage_account_name:
+            # Extract the resource group name from the account ID
+            # An example of account.id would be the following:
+            # /subscriptions/{subscription_id}/resourceGroups/{resource_group_name}/providers/Microsoft.Storage/storageAccounts/{container_name} # pylint: disable=line-too-long
+            resource_group_name = account.id.split('/')[4]
+            return resource_group_name
+    # resource group cannot be found when using container not created
+    # under the user's subscription id, i.e. public container or
+    # private containers not belonging to the user.
+    return None
+
+
+def get_az_storage_account_key(
+    storage_account_name: str,
+    resource_group_name: Optional[str] = None,
+    storage_client: Optional[Client] = None,
+    resource_client: Optional[Client] = None,
+) -> Optional[str]:
+    """Returns access key of the given name of storage account.
+
+    Args:
+        storage_account_name: str; Name of the storage account
+        resource_group_name: str; Name of the resource group the passed storage
+            account belongs to.
+        storage_clent: Optional[Client]; client object facing Storage
+        resource_client: Optional[Client]; client object facing Resource
+
+    Returns:
+        One of the few access keys to the given storage account
+    """
+    # resource_group_name is None when using a public container or
+    # a private containers not belonging to the user.
+    if resource_group_name is None:
+        return None
+    if resource_client is None:
+        resource_client = create_az_client('resource')
+    if storage_client is None:
+        storage_client = create_az_client('storage')
+    resources = resource_client.resources.list_by_resource_group(
+        resource_group_name)
+    assert resources is not None
+    for resource in resources:
+        if (resource.type == 'Microsoft.Storage/storageAccounts' and
+                resource.name == storage_account_name):
+            storage_account_keys = storage_client.storage_accounts.list_keys(
+                resource_group_name, storage_account_name)
+            storage_account_keys = [
+                key.value for key in storage_account_keys.keys
+            ]
+    storage_account_key = storage_account_keys[0]
+    return storage_account_key
+
+
+def get_az_container_sas_token(
+    storage_account_name: str,
+    storage_account_key: str,
+    container_name: str,
+) -> str:
+    """Returns SAS token used to access container.
+
+    Args:
+        storage_account_name: str; Name of the storage account
+        storage_account_key: str; access key for the given storage
+            account
+        container_name: str; name of the mounting container
+
+    Returns:
+        SAS token prepended with the delimiter character, "?"
+    """
+    from azure.storage.blob import generate_container_sas, ContainerSasPermissions
```

Comment:
should we move this into `sky/adaptors/azure.py`? Since here we import some modules from azure core.

Reply:
Done